src/main/java/org/jenkinsci/plugins/github_branch_source/Connector.java

```diff
@@ -580,15 +584,17 @@ static void checkApiRateLimit(@NonNull TaskListener listener, GitHub github)
 
         @Override
         public long getRecurrencePeriod() {
-            return TimeUnit.MINUTES.toMillis(2);
+            return TimeUnit.MINUTES.toMillis(5);
         }
 
         @Override
         protected void doRun() throws Exception {
-            // free any connection unused for the last 5 minutes
-            long threshold = System.currentTimeMillis() - TimeUnit.MINUTES.toMillis(5);
+            // Free any connection that is unused (zero refs)
+            // and has not been looked up or released for the last 30 minutes
+            long unusedThreshold = System.currentTimeMillis() - TimeUnit.MINUTES.toMillis(30);
```

Comment:
Longer unused threshold means even if this is not completely fixed it will be much less likely to happen.  
This also means there are likely to be some extra cache folders present for a little while after they go stale. This is not great, but it is better than the alternative.